neural_lam/utils.py

```diff
@@ -260,11 +265,50 @@ def fractional_plot_bundle(fraction):
     return bundle
 
 
-def init_wandb_metrics(wandb_logger, val_steps):
-    """
-    Set up wandb metrics to track
-    """
-    experiment = wandb_logger.experiment
+@rank_zero_only
+def rank_zero_print(*args, **kwargs):
+    """Print only from rank 0 process"""
+    print(*args, **kwargs)
+
+
+@rank_zero_only
+def init_wandb(args):
+    """Initialize wandb"""
+    if args.resume_run is None:
+        prefix = f"subset-{args.subset_ds}-" if args.subset_ds else ""
+        if args.eval:
+            prefix = prefix + f"eval-{args.eval}-"
+        random_int = random.randint(0, 10000)
+        run_name = (
+            f"{prefix}{args.model}-{args.processor_layers}x{args.hidden_dim}-"
+            f"{time.strftime('%m_%d_%H_%M_%S')}-{random_int}"
+        )
+        wandb.init(
+            name=run_name,
+            project=args.wandb_project,
+            config=args,
+        )
+        logger = pl.loggers.WandbLogger(
```

Comment:
Now we get a warning every time the WandbLogger is created, as wandb is already initialized. Is it really necessary to first do the `wandb.init` call? Would the same thing not be achieved by just creating the `WandbLogger`? (and not have that warning) `WandbLogger` should be able to take all the same arguments as `wandb.init`.